common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java

```diff
@@ -377,12 +379,155 @@ protected void renewAccessToken(@NonNull final SilentTokenCommandParameters para
+                    throw new ServiceException(SERVICE_NOT_AVAILABLE, "AAD is not available.", tokenResult.getErrorResponse().getStatusCode(), null);
+                }
+
             } else {
                 Logger.warn(TAG, "Invalid state, No token success or error response on the token result");
             }
         }
     }
 
+    /*
+        Light version of:
+        protected renewAccessToken(@NonNull final SilentTokenCommandParameters parameters,
+                                   @NonNull final AcquireTokenResult acquireTokenSilentResult,
+                                   @SuppressWarnings(WarningType.rawtype_warning) @NonNull final OAuth2TokenCache tokenCache,
+                                   @SuppressWarnings(WarningType.rawtype_warning) @NonNull final OAuth2Strategy strategy,
+                                   @NonNull final ICacheRecord cacheRecord)
+
+        Diffs
+        1) acquireTokenSilentResult updates omitted.
+        2) All arguments derived from SilentTokenCommandParameters
+        3) New logic replacing old Access Token
+     */
+    public TokenResult renewAccessToken(@NonNull final SilentTokenCommandParameters parameters)
+            throws IOException, ClientException, ServiceException {
+        final String methodName = ":renewAccessToken";
+        Logger.info(
+                TAG + methodName,
+                "Renewing access token..."
+        );
+
+        OAuth2Strategy strategy = getStrategy(parameters);
+        OAuth2TokenCache cache = getTokenCache(parameters);
+        ICacheRecord cacheRecord = getCacheRecord(parameters);
+
+        Logger.info(
+                TAG + methodName,
+                "Attempting renewal of Access Token because it's refresh-expired. RefreshIn was expired at " + cacheRecord.getAccessToken().getRefreshOn() + ". Regular expiry is at " + cacheRecord.getAccessToken().getExpiresOn() + "."
+                        + "Currently executing acquireTokenSilent(..), SilentTokenCommand with CorrelationId: " + parameters.getCorrelationId()
+        );
+        //Get tokenResult
+        RefreshTokenRecord refreshTokenRecord = cacheRecord.getRefreshToken();
+        logParameters(TAG, parameters);
+        final TokenResult tokenResult = performSilentTokenRequest(
+                strategy,
+                refreshTokenRecord,
+                parameters
+        );
+
+        logResult(TAG + methodName, tokenResult);
+        if (tokenResult.getSuccess()) {
+            Logger.info(
+                    TAG + methodName,
+                    "Token request was successful"
+            );
+
+            // Remove old Access Token
+            Logger.info(
+                    TAG + methodName,
+                    "Access token is refresh-expired. Removing from cache..."
+            );
+            final AccessTokenRecord accessTokenRecord = cacheRecord.getAccessToken();
+            cache.removeCredential(accessTokenRecord);
```

Comment:
Why do we need to do this? (Aren't we going to be overwriting the token below?)

Reply:
We can leave it out if you prefer, it's added as a redundancy here.